azurerm/internal/services/servicefabricmesh/resource_arm_service_fabric_mesh_application.go

```diff
@@ -0,0 +1,436 @@
+				return fmt.Errorf("checking for presence of existing service fabric mesh application: %+v", err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_service_fabric_mesh_application", *existing.ID)
+		}
+	}
+
+	parameters := servicefabricmesh.ApplicationResourceDescription{
+		ApplicationResourceProperties: &servicefabricmesh.ApplicationResourceProperties{
+			Services: expandServiceFabricMeshApplicationServices(d.Get("service").(*schema.Set).List()),
+		},
+		Location: utils.String(location),
+		Tags:     tags.Expand(t),
+	}
+
+	if _, err := client.Create(ctx, resourceGroup, name, parameters); err != nil {
+		return fmt.Errorf("creating Service Fabric Mesh Application %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	log.Printf("[DEBUG] Waiting for Service Fabric Mesh Application %q (Resource Group %q) to finish creating", name, resourceGroup)
+	stateConf := &resource.StateChangeConf{
+		Pending:                   []string{string(servicefabricmesh.Creating), string(servicefabricmesh.Upgrading), string(servicefabricmesh.Deleting)},
+		Target:                    []string{string(servicefabricmesh.Ready)},
+		Refresh:                   serviceFabricMeshApplicationCreateRefreshFunc(ctx, client, resourceGroup, name),
+		MinTimeout:                10 * time.Second,
+		ContinuousTargetOccurence: 10,
+		Timeout:                   d.Timeout(schema.TimeoutCreate),
+	}
+
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("waiting for Service Fabric Mesh Application %q (Resource Group %q) to become available: %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("retrieving Service Fabric Mesh Application %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmServiceFabricMeshApplicationRead(d, meta)
+}
+
+func resourceArmServiceFabricMeshApplicationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ServiceFabricMesh.ApplicationClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ServiceFabricMeshApplicationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Unable to find Service Fabric Mesh Application %q - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("reading Service Fabric Mesh Application: %+v", err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+
+	/* This does not currently return from the api
+	if err := d.Set("service", flattenServiceFabricMeshApplicationServices(resp.Services)); err != nil {
+		return fmt.Errorf("setting `service`: %+v", err)
+	}*/
```

Comment:
should this be commented out?

Reply:
It should. These attributes aren't returned from the api at the moment.

Reply:
Looks like this might be available using the `ServicesClient`?

Reply:
Boom, that's it. Thank you @jackofallops!